src/CoreWCF.Http/tests/MessageEncodingTests.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using CoreWCF.Configuration;
+using Helpers;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace CoreWCF.Http.Tests
+{
+    public class MessageEncodingTests
+    {
+        private readonly ITestOutputHelper _output;
+
+        public MessageEncodingTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        public static IEnumerable<object[]> GetTestVariations()
+        {
+            yield return new object[] { typeof(BasicHttpBindingWithTextMessageEncodingStartup), BasicHttpBindingWithTextMessageEncodingStartup.GetClientBinding() };
+            yield return new object[] { typeof(BasicHttpBindingWithMtomMessageEncodingStartup), BasicHttpBindingWithMtomMessageEncodingStartup.GetClientBinding() };
+            yield return new object[] { typeof(WSHttpBindingWithTextMessageEncodingStartup), WSHttpBindingWithTextMessageEncodingStartup.GetClientBinding() };
+            yield return new object[] { typeof(WSHttpBindingWithMtomMessageEncodingStartup), WSHttpBindingWithMtomMessageEncodingStartup.GetClientBinding() };
+            yield return new object[] { typeof(NetHttpBindingWithTextMessageEncodingStartup), NetHttpBindingWithTextMessageEncodingStartup.GetClientBinding() };
+            yield return new object[] { typeof(NetHttpBindingWithMtomMessageEncodingStartup), NetHttpBindingWithMtomMessageEncodingStartup.GetClientBinding() };
+            yield return new object[] { typeof(NetHttpBindingWithBinaryMessageEncodingStartup), NetHttpBindingWithBinaryMessageEncodingStartup.GetClientBinding() };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTestVariations))]
+        public void EchoByteArray(Type startupType, System.ServiceModel.Channels.Binding binding)
+        {
+            IWebHost host = ServiceHelper.CreateWebHostBuilder(_output, startupType).Build();
+            using (host)
+            {
+                host.Start();
+                var factory = new System.ServiceModel.ChannelFactory<ClientContract.IMessageEncodingService>(binding,
+                    new System.ServiceModel.EndpointAddress(new Uri("http://localhost:8080/MessageEncodingService/IMessageEncodingService.svc")));
+                ClientContract.IMessageEncodingService channel = factory.CreateChannel();
+                byte[] bytes = new byte[768];
+#if NET472_OR_GREATER
+                using (RNGCryptoServiceProvider rng = new RNGCryptoServiceProvider())
+                {
```

Comment:
I can see this being a common thing to use in tests, can you refactor this into the one of the helper classes to a method which returns a random byte array of the specified size.